pkg/auth/authenticator/bearertoken/bearertoken.go

```diff
@@ -20,16 +20,19 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/go-openapi/spec"
```

Comment:
@ericchiang pkg/apiserver/authenticator is not the only place that uses authenticators (k8s and federation server.go are at least two other places I see this) so the logic will be fragmented into those places authenticators are used and it will be in any place we use authenticators (now or later). 

I am trying to understand your (also @liggitt) concern here, is it the place of the logic or the change in interfaces or dependencies? I may be able fix the interfaces (no change in interfaces) but keep the logic in authenticators.

@nikhiljindal That is not possible until we depend on switch case on type of authenticators and have access to their internals by adding access methods because of union and wrapper authenticators. If I did not understand what you are suggesting, please explain.